marketdata/rest_test.go

```diff
@@ -13,6 +14,84 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestDefaultDo(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, `{"symbol":"SPY","bar":{"t":"2021-11-20T00:59:00Z","o":469.18,"h":469.18,"l":469.11,"c":469.17,"v":740,"n":11,"vw":469.1355}}`)
+	}))
+	defer server.Close()
+	client := NewClient(ClientOpts{
+		BaseURL: server.URL,
+	})
+	bar, err := client.GetLatestBar("SPY")
+	require.NoError(t, err)
+	assert.Equal(t, 469.11, bar.Low)
+}
+
+func TestDefaultDo_InternalServerError(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+	}))
+	defer server.Close()
+	originalDataURL := os.Getenv("APCA_API_DATA_URL")
+	defer func() { os.Setenv("APCA_API_DATA_URL", originalDataURL) }()
+	require.NoError(t, os.Setenv("APCA_API_DATA_URL", server.URL))
+	client := NewClient(ClientOpts{
+		OAuth: "myoauthkey",
+	})
+	_, err := client.GetLatestBar("SPY")
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "500")
+}
+
+func TestDefaultDo_One429(t *testing.T) {
```

Comment:
We are testing retry functionality here, maybe the name could be _Retry instead of _One429.